lib/Core/Site/Pagination/Pagerfanta/ContentSearchHitAdapter.php

```diff
@@ -93,18 +93,13 @@ public function getSlice($offset, $length)
         $query->limit = $length;
         $query->performCount = false;
 
-        $this->searchResult = $this->findService->findContent($query);
+        $searchResult = $this->findService->findContent($query);
 
-        // Set count for further use if returned by search engine despite !performCount (Solr, ES)
-        if (!isset($this->nbResults) && isset($this->searchResult->totalCount)) {
```

Comment:
I think the problem was in setting the search result to the property, if that is removed shouldn't it be possible to keep this?

Reply:
@mivancic I think setting the facets was added for the convenience, so you don't have to fetch the entire results page to retrieve facets. For instance, instead of doing
$pager->getCurrentPageResults();
$facets = $pager->adapter->getFacets();

, you can simply call a count method without having to retrieve the results, and have facets provided to you.
But I am not 100% sure. @pspanja WTYT?

Reply:
Yes, but but still you shouldn't have to call `$pager->getCurrentPageResults()` just to later get facets from `$pager->adapter->getFacets()`, that should work standalone.

AFAIR the problem was caching the result here in `$this->searchResult`, and this with `$query->performCount = false`. So if you accessed the result first, `nbResults` would be `null`, but if you access `nbResults` first, everything is in order. That's why I think it should be possible to keep setting `ngResults` and `facets` here -- since we are not setting `$this->searchResult` any more.

Reply:
@pspanja fair point. I think we had an issue or a pull request opened earlier because of the same reason, right?

Reply:
> Same as the totalCount, the facets have global value and do not depend on the current result.

Yes, exactly. It was probably a mistake to include this in the release, deprecating it at the same moment, providing new adapters. But oh well :)

Reply:
@hknezevic hum, not sure, I can't find one.